src/test/java/nl/stil4m/mollie/concepts/CustomerSubscriptionsIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+public class CustomerSubscriptionsIntegrationTest {
+
+    private CustomerPayments customerPayments;
+    private CustomerSubscriptions customerSubscriptions;
+
+    @Before
+    public void before() throws InterruptedException, IOException {
+        Thread.sleep(TEST_TIMEOUT);
+        Client client = strictClientWithApiKey(VALID_API_KEY);
+
+        String uuid = UUID.randomUUID().toString();
+        Map<String, Object> defaultMetadata = new HashMap<>();
+        defaultMetadata.put("foo", "bar");
+        defaultMetadata.put("id", uuid);
+
+        String name = "Test Customer " + uuid;
+        Customer customer = client.customers().create(new CreateCustomer(name, uuid + "@foobar.com", Optional.empty(), defaultMetadata)).getData();
+
+        customerPayments = client.customerPayments(customer.getId());
+        customerSubscriptions = client.customerSubscriptions(customer.getId());
+
+        //setup selenium
+        ChromeDriverManager.getInstance().setup();
+    }
+
+    @Test
+    public void testList() throws IOException, URISyntaxException {
+        ResponseOrError<Page<Subscription>> list = customerSubscriptions.list(Optional.empty(), Optional.empty());
+
+        assertThat(list.getSuccess(), is(true));
+    }
+
+    @Test
+    public void testCreate() throws IOException, URISyntaxException, InterruptedException {
+        // https://www.mollie.com/en/docs/reference/subscriptions/create
+        //create payment that creates membate (required to create subscription), this is kinda a confirm for the payment info to do the recurring payment over
+        CreatePayment createIdealPayment = new CreateIdealPayment(10d, "test payment for subscription", "https://example.com/thank/you", Optional.empty(), null, null);
+        FirstRecurringPayment customerPayment = new FirstRecurringPayment(createIdealPayment);
+        Payment payment = customerPayments.create(customerPayment).getData();
+
+        String paymentUrl = payment.getLinks().getPaymentUrl();
+
+        //open paymentUrl and set it to paid
+
+        //setup selenium
+        WebDriver driver = new ChromeDriver();
+
+        driver.get(paymentUrl);
+        // Find ok button and click it
+        driver.findElement(By.name("issuer")).click();
+
+        //find "paid" radiobutton and click it
+        driver.findElements(By.name("final_state")).forEach(radioButton -> {
+            if (radioButton.getAttribute("value").equals("paid")) {
+                radioButton.click();
+            }
+        });
+
+        //submit form
+        driver.findElement(By.cssSelector("#footer > button")).click();
+
+        driver.quit();
+
+        //check if payment is complete
+        payment = customerPayments.get(payment.getId()).getData();//get new version of payment
+        if (!payment.getStatus().equals("paid")) {
```

Comment:
Would it be a good interface to add a `isPaid()` method to payment? (which will be ignored in the JSON serialization)

Reply:
I don't agree to just add `isPaid()`, because a payment can have 9 types of statuses (source:  https://www.mollie.com/en/docs/status)
I would suggest making it an enum, what do you think about that?